core/src/main/java/org/axonframework/messaging/Message.java

```diff
@@ -91,4 +93,28 @@
      * @return a copy of this message with the given MetaData
      */
     Message<T> andMetaData(Map<String, ?> metaData);
+
+    /**
+     * Serialize the payload of this message using given {@code serializer}, using given
+     * {@code expectedRepresentation}. This method <em>should</em> return the same SerializedObject instance when
+     * invoked multiple times using the same serializer.
+     *
+     * @param serializer             The serializer to serialize payload with
+     * @param expectedRepresentation The type of data to serialize to
+     * @param <R>                    The type of data to serialize to
+     * @return a SerializedObject containing the serialized representation of the message's payload
+     */
+    <R> SerializedObject<R> serializePayload(Serializer serializer, Class<R> expectedRepresentation);
+
+    /**
+     * Serialize the meta data of this message using given {@code serializer}, using given
```

Comment:
Suggestion: I'd rewrite the first sentence to:
`Serialize the meta data of this message to the {@code expectedRepresentation} using the given {@link Serializer}.`